src/App.tsx

```diff
@@ -1,26 +1,224 @@
+    todosFromServer
+      .getTodos()
+      .then(setTodos)
+      .catch(() => setLoadError(true));
+    wait(3000).then(() => setLoadError(false));
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const deleteCompletedTodos = (todosToDelete: Todo[]) => {
+    todosToDelete.forEach(todo => deleteTodo(todo));
+  };
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">
-          React Todo App - Add and Delete
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {/* Add a todo on form submit */}
+          <TodoForm
+            onSubmit={addTodo}
+            setTitleError={setTitleError}
+            todos={todos}
+            updateTodo={updateTodo}
+            setTempArray={temp}
+            edit={edit}
+          />
+        </header>
+
+        <TodoList
+          todos={filteredTodos}
+          updateTodo={updateTodo}
+          deleteTodo={deleteTodo}
+          array={loadingTodos}
+          setLoadingTodos={temp}
+          edit={edit}
+        />
+
+        {tempTodo && (
+          <TodoItem
+            todo={tempTodo}
+            updateTodo={updateTodo}
+            deleteTodo={deleteTodo}
+            loadingTodos={loadingTodos}
+            setLoadingTodos={temp}
+            edit={edit}
+          />
+        )}
+
+        {!!todos.length && (
+          // {/* Hide the footer if there are no todos */}
+          <TodoFooter
+            todos={todos}
+            setStatus={setStatus}
+            status={status}
+            deleteCompletedTodos={deleteCompletedTodos}
+            setLoadingTodos={setLoadingTodos}
+          />
+        )}
+      </div>
+
+      {/* {error && ( */}
+      {/* DON'T use conditional rendering to hide the notification */}
+      {/* Add the 'hidden' class to hide the message smoothly */}
+      <div
+        data-cy="ErrorNotification"
+        className={classNames(
+          'notification is-danger is-light has-text-weight-normal',
+          {
+            hidden:
+              !titleError &&
+              !loadError &&
+              !addError &&
+              !deleteError &&
+              !updateError,
+          },
+        )}
+      >
+        <button data-cy="HideErrorButton" type="button" className="delete" />
+        {/* show only one message at a time */}
+        {loadError && 'Unable to load todos'}
+        {titleError && 'Title should not be empty'}
+        {addError && 'Unable to add a todo'}
+        {deleteError && 'Unable to delete a todo'}
+        {updateError && 'Unable to update a todo'}
```

Comment:
can we just create only one errorMessage state and save our messages there?
```suggestion
        {errorMessage}
```
I think it should work